portality/tasks/harvester.py

```diff
@@ -0,0 +1,128 @@
+        if self.mail_prereqs:
+            self.mail.send_mail(
+                to=app.config["HARVESTER_EMAIL_RECIPIENTS"],
+                fro=self.fro,
+                subject=self.sub_prefix + "DOAJ Harvester finished at {0}".format(
+                    datetime.datetime.utcnow().strftime("%Y-%m-%dT%H:%M:%SZ")),
+                msg_body=report
+            )
+
+            return
+
+    def cleanup(self):
+        """
+        Cleanup after a successful OR failed run of the task
+        :return:
+        """
+        pass
+
+    @classmethod
+    def prepare(cls, username, **kwargs):
+        """
+        Take an arbitrary set of keyword arguments and return an instance of a BackgroundJob,
+        or fail with a suitable exception
+
+        :param kwargs: arbitrary keyword arguments pertaining to this task type
+        :return: a BackgroundJob instance representing this task
+        """
+
+        # run_only_once()
+        # initialise_index(app, es_connection)
+        cls.sub_prefix = app.config.get('HARVESTER_EMAIL_SUBJECT_PREFIX', '')
+
+        # Send an email when the harvester starts.
+        cls.mail_prereqs = False
+        cls.fro = app.config.get("HARVESTER_EMAIL_FROM_ADDRESS", 'harvester@doaj.org')
+        if app.config.get("HARVESTER_EMAIL_ON_EVENT", False):
+            to = app.config.get("HARVESTER_EMAIL_RECIPIENTS", None)
+
+            if to is not None:
+                cls.mail_prereqs = True
+                from portality import app_email as mail
+                mail.send_mail(
+                    to=to,
+                    fro=cls.fro,
+                    subject=cls.sub_prefix + "DOAJ Harvester started at {0}".format(
+                        datetime.datetime.utcnow().strftime("%Y-%m-%dT%H:%M:%SZ")),
+                    msg_body="A new running instance of the harvester has started."
+                )
+
+        if app.debug:
+            # Augment the default flask debug log to include a timestamp.
+            app.debug_log_format = (
+                    '-' * 80 + '\n' +
+                    '%(asctime)s\n'
+                    '%(levelname)s in %(module)s [%(pathname)s:%(lineno)d]:\n' +
+                    '%(message)s\n' +
+                    '-' * 80
+            )
+            flask.logging.create_logger(app)
+
+        # first prepare a job record
+        job = models.BackgroundJob()
+        job.user = username
+        job.action = cls.__action__
+        return job
+
+    @classmethod
+    def submit(cls, background_job):
+        """
+        Submit the specified BackgroundJob to the background queue
+
+        :param background_job: the BackgroundJob instance
+        :return:
+        """
+        background_job.save()
+        harvest.schedule(args=(background_job.id,), delay=10)
+        # fixme: schedule() could raise a huey.exceptions.HueyException and not reach redis- would that be logged?
+
+
+@main_queue.periodic_task(schedule("harvest"))
```

Comment:
suggest that we put this on the `long_running` queue instead, as the harvester can run for a while